### PR TITLE
Add `Global_context`.

### PR DESCRIPTION
**Pull request by lhmouse**

### CODE REVIEW

File: asteria/src/global_context.cpp [hunk 1]
```diff
@@ -0,0 +1,23 @@
+// This file is part of Asteria.
+// Copyleft 2018, LH_Mouse. All wrongs reserved.
+
+#include "precompiled.hpp"
+#include "global_context.hpp"
+#include "utilities.hpp"
+
+namespace Asteria {
+
+Global_context::~Global_context()
```
Comment (MaskRay) [inline asteria/src/global_context.cpp]: `= default` and `{}` are probably the same (they are not the first declarations and the destructors are non-trivial anyway) but `{}` takes you two more lines.

File: asteria/src/global_context.hpp [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+#ifndef ASTERIA_GLOBAL_CONTEXT_HPP_
+#define ASTERIA_GLOBAL_CONTEXT_HPP_
+
+#include "fwd.hpp"
+#include "abstract_context.hpp"
+
+namespace Asteria {
+
+class Global_context : public Abstract_context
```
Comment (MaskRay) [inline asteria/src/global_context.hpp]: Just a question: have you designed global context with dynamic-scoped variables or thread_local in mind?

File: asteria/src/global_context.cpp (hunk 1 above)
Comment (lhmouse) [reply asteria/src/global_context.cpp]: Yes, as there is no difference, it is reasonable to prefer `{ }` to `= default` which is less universal, not to mention that destructors get implicit `noexcept(true)` (we do not make use of throwing anywhere).

File: asteria/src/global_context.hpp (hunk 2 above)
Comment (lhmouse) [reply asteria/src/global_context.hpp]: The global context is used to share data between **different files**. There is no syntactical construction to declare a new named reference in it. As a result, references in the global context can only be created via binding functions. Its main purpose was the `import` directive which was only barely in my mind and now (I think) is superfluous. The `require` operator (like in _Node.js_ and _Lua_) would be preferred, which, nevertheless, still requires a global context in order to detect circular dependencies.

**Pull request merged**